### ISSUE TITLE
Why popup.options.offset does not reflect popupAnchor any more?

### ISSUE THREAD

**Issue by remilev** [Leaflet/Leaflet#4888]
Hi, I have several custom icons with different popupAnchor values. I declare them with either `L.Icon.extend` or `L.DivIcon.extend`.

When executing `map.on('popupopen', function(e) { console.log(e.popup.options.offset) });`

I get Object {x: -3, y: -69} with leaflet v1beta2 and Array [0, 7] with leaflet RC3 (for the same popup)

- popup.options.offset.y or x are undefined with RC3
- with leaflet v1beta2, popup.options.offset vary from one popup to another (linked to anchor value) whereas with RC3 popup.options.offset doesn't change (doesn't "mirror" anchor value)

Is that intended?

note : with RC3, I then must do : 
`map.on('popupopen', function(e) { console.log(e.popup._source.options.icon.options.popupAnchor) });`

maybe related to : https://github.com/Leaflet/Leaflet/pull/4662
Is this fix the reason?

**Comment by yohanboniface** [comment]
> Is that intended?

Yes, altering the options during runtime is considered a bad practice. And this was causing troubles when updating an icon.

>note : with RC3, I then must do :
map.on('popupopen', function(e) { console.log(e.popup._source.options.icon.options.popupAnchor) });

@mourner @IvanSanchez @perliedman what about adding `source` key to the popup event (which may or not be defined whether the popup has a source or not)? That may help in such scenarios.
